src/libraries/System.IO.Compression/src/System/IO/Compression/ZipArchiveEntry.cs

```diff
@@ -87,9 +86,11 @@ internal ZipArchiveEntry(ZipArchive archive, ZipCentralDirectoryFileHeader cd)
 
             _fileComment = cd.FileComment;
 
-            _compressionLevel = MapCompressionLevel(_generalPurposeBitFlag, CompressionMethod);
-
             Changes = ZipArchive.ChangeState.Unchanged;
+
+            // Setting CompressionMethod can change the _versionToExtract variable, which can change the value of Changes
+            CompressionMethod = (CompressionMethodValues)cd.CompressionMethod;
+            _compressionLevel = MapCompressionLevel(_generalPurposeBitFlag, CompressionMethod);
```

Comment:
Is there a motive to move these two together to this location? I'm failing to see one, but maybe I'm wrong. Usually we try to minimize changes.

Reply:
There is. The setter for `CompressionMethod` calls `VersionToExtractAtLeast`, which now sets `Changes`, so I have to set CompressionMethod after Changes. I've taken another look and reduced the diff slightly though.

Reply:
Ok got it, thanks! I would've been fine with your initial change after you shared the motive, but no need to revert it again. This is good too.